File: samples/CS/04-TabGroupHelp/RibbonItems.cs
using System;
using System.Windows.Forms;

namespace WinForms.Ribbon
{
    partial class RibbonItems
    {
        private Form _form;

        public void Init(Form form)
        {
            _form = form;
            ButtonExit.Click += new EventHandler<EventArgs>(_exitButton_ExecuteEvent);
            HelpButton.Click += new EventHandler<EventArgs>(_helpButton_ExecuteEvent);
        }

        void _exitButton_ExecuteEvent(object sender, EventArgs e)
        {
            // Close form asynchronously since we are in a ribbon event 
            // handler, so the ribbon is still in use, and calling Close 
            // will eventually call _ribbon.DestroyFramework(), which is 
            // a big no-no, if you still use the ribbon.
            _form.BeginInvoke(new MethodInvoker(_form.Close));
        }

        void _helpButton_ExecuteEvent(object sender, EventArgs e)
        {
            MessageBox.Show("Help button pressed");
        }

        public void Load()
        {
        }

    }
}

File: samples/VB/04-TabGroupHelp/RibbonItems.vb
Imports System
Imports System.Windows.Forms
Imports _04_TabGroupHelp

Namespace WinForms.Ribbon

    Partial Class RibbonItems
		Private _self As Form
		Public Sub Init(self As Form)
			_self = self

			AddHandler ButtonExit.Click, AddressOf _exitButton_ExecuteEvent
			AddHandler _HelpButton.Click, AddressOf _helpButton_ExecuteEvent
		End Sub


		Private Sub _exitButton_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			' Close form asynchronously since we are in a ribbon event 
			' handler, so the ribbon is still in use, and calling Close 
			' will eventually call _ribbon.DestroyFramework(), which is 
			' a big no-no, if you still use the ribbon.
			_self.BeginInvoke(New MethodInvoker(AddressOf _self.Close))
		End Sub

		Private Sub _helpButton_ExecuteEvent(ByVal sender As Object, ByVal e As EventArgs)
			MessageBox.Show("Help button pressed")
		End Sub

	End Class
End Namespace
